core/src/identity.rs

```diff
@@ -0,0 +1,226 @@
+                    "id":"did:knox:zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ#zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ",
+                    "master_public_key":"zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ",
+                    "proof_type":"Ed25519Signature2018",
+                    "public_key_multibase":"CapabilityDelegation"
+                }],
+                "capability_invocation":[{
+                    "controller":"did:knox:zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ",
+                    "id":"did:knox:zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ#zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ",
+                    "master_public_key":"zFCxaFZ4twBFG8P2hBvzheaRdsSshqEngn9r4nuQwEMfJ",
+                    "proof_type":"Ed25519Signature2018",
+                    "public_key_multibase":"CapabilityInvocation"
+                }],
+                "context":["default"],
+                "id":"default"
+            })
+    }
+
+    fn get_did() -> String {
+        String::from("123456789")
+    }
+
+    #[rstest::rstest]
+    #[case::created_successfully(
+        Some(Ok(())),
+        get_did(),
+        get_json_input_mock(),
+        true
+    )]
+    #[case::created_error(
+        Some(Err(crate::error::ResolverError{
+            message: "testErr".to_string(), 
+            kind: crate::error::ErrorKind::NetworkFailure})),
+        get_did(),
+        get_json_input_mock(),
+        false
+    )]
+
+    fn test_create_identity(
+        #[case] mock_create_response: Option<Result<(), crate::error::ResolverError>>,
+        #[case] did_doc: String,
+        #[case] did_document_input_mock: serde_json::Value,
+        #[case] expect_ok: bool,
+    ) -> Result<(), String> {
+        let mut resolver_mock = MockDIDResolver::default();
+
+        resolver_mock
+            .expect_create()
+            .with(
+                mockall::predicate::function(|did_doc: &String| -> bool {
+                    did_doc.clone().len() > 0
+                }),
+                mockall::predicate::function(move |doc_input: &serde_json::Value| -> bool {
+                    let did_doc_input: DidDocument =
+                        serde_json::from_value(doc_input.clone()).unwrap();
+                    let did_doc_mock: DidDocument =
+                        serde_json::from_value(did_document_input_mock.clone()).unwrap();
+
+                    if did_doc_input.id != did_doc_mock.id {
+                        return false;
+                    }
+                    if did_doc_input.context != did_doc_mock.context {
+                        return false;
+                    }
+                    if did_doc_input.assertion_method[0].public_key_multibase
+                        != did_doc_mock.assertion_method[0].public_key_multibase
+                    {
+                        return false;
+                    }
+                    return true;
+                }),
```

Comment:
these can be simplified into one statement: `return did_doc_input.id == did_doc_mock.id && did_doc_input.context == did_doc_mock.context && ...;`